src/structures/CommandInteractionOptionResolver.js

```diff
@@ -239,6 +239,18 @@ class CommandInteractionOptionResolver {
     const option = this._getTypedOption(name, '_MESSAGE', ['message'], required);
     return option?.message ?? null;
   }
+
+  /**
+   * Gets a focused option.
+   * @param {name} name The name of the option.
+   * @param {boolean} [required=false] Whether to throw an error if the option is not found.
+   * @returns {?(string)}
+   * The value of the option, or null if not set and not required.
+   */
+  getFocused(name, required = false) {
+    const option = this._getTypedOption(name, 'STRING', ['value'], required);
```

Comment:
The option is not always going to be a string type. It can be any type that accepts the choices / autocomplete parameter

Reply:
I don't see any reason to make this an option you have to get by name either. There can only ever be one focused option, so I'd design it more like `getSubcommand()` which can only be singular.